components/style/matching.rs

```diff
@@ -769,22 +769,27 @@ pub trait MatchMethods : TElement {
                 //
```

Comment:
I guess? I don't think that's a strict improvement though. This is complex just because in gecko `old_style` != old style source, and I think that's a complexity we need to deal with anyway.